frontend/src/static/js/components/webstatus-notfound-error-page.ts

```diff
@@ -50,75 +98,137 @@ export class WebstatusNotFoundErrorPage extends LitElement {
-          display: flex;
+        .similar-results-header {
+          color: #1a1a1a;
+          font-weight: 500;
+          margin-bottom: 6px;
         }
-
-        #error-action-report a {
-          color: inherit;
+        .feature-list {
+          list-style: none;
+          padding: 0;
+          margin: 0;
+        }
+        .feature-list li {
+          padding: 6px 0;
+        }
+        .feature-list li a {
           text-decoration: none;
+          color: #007bff;
+          font-weight: 500;
+        }
+        .feature-list li a:hover {
+          text-decoration: underline;
+          color: #0056b3;
         }
       `,
     ];
   }
+
   protected render(): TemplateResult {
+    const featureId = getSearchQuery(this.location);
+
+    console.log('hello', featureId);
+
     return html`
-      <div id="error-container">
+      <div
+        id="error-container"
+        style="gap: ${this.similarFeatures.length > 0 ? '32px' : '48px'};"
+      >
         <div id="error-header">
           <div id="error-status-code">404</div>
           <div id="error-headline">Page not found</div>
           <div id="error-detailed-message">
-            We couldn't find the page you're looking for.
+            ${featureId
+              ? html`
+                  We could not find Feature ID: <strong>${featureId}</strong>
+                `
+              : html`<span class="error-message"
+                  >We couldn't find the page you're looking for.</span
+                >`}
           </div>
         </div>
 
-        <div id="error-actions">
-          <div id="error-action-home">
-            <sl-button id="error-action-home-btn" variant="primary" href="/"
-              >Go back home</sl-button
-            >
-          </div>
-          <div id="error-action-report">
+        ${featureId
+          ? html`
+              ${this._loadingSimilarResults?.status === TaskStatus.PENDING
+                ? html`<p class="loading-message">
+                    Loading similar features...
+                  </p>`
+                : this.similarFeatures.length > 0
+                  ? html`
+                      <div class="similar-features-container">
+                        <p class="similar-results-header">
+                          Here are some similar features:
+                        </p>
+                        <ul class="feature-list">
+                          ${this.similarFeatures.map(
+                            f =>
+                              html`<li><a href="${f.url}">${f.name}</a></li>`,
+                          )}
+                        </ul>
+                      </div>
+                    `
+                  : html`<p class="error-message">
+                      No similar features found.
+                    </p>`}
+            `
+          : ''}
+
+        <div
+          id="error-actions"
+          style="gap: ${featureId && this.similarFeatures.length > 0
+            ? '16px'
```

Comment:
We should not inline styles. Instead have a different class for each case. and assign the class appropriately